plugins/module_utils/network/sonic/config/vrrp/vrrp.py

```diff
@@ -0,0 +1,686 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2022 Dell Inc. or its subsidiaries. All Rights Reserved
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The sonic_vrrp class
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.cfg.base import (
+    ConfigBase,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    to_list,
+)
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.facts.facts import Facts
+
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.utils.utils \
+    import (
+        get_diff,
+        update_states,
+    )
+
+from ansible_collections.dellemc.enterprise_sonic.plugins.module_utils.network.sonic.sonic import (
+    to_request,
+    edit_config
+)
+
+PATCH = 'patch'
+DELETE = 'delete'
+
+ipv4_path = '/openconfig-if-ip:ipv4/addresses/address=1.1.1.1/'
+ipv6_path = '/openconfig-if-ip:ipv6/addresses/address=1::1/'
```

Comment:
Why are these addresses hardcoded?

Reply:
The VRRP REST API needs an IP address as default in the URL. On comparing with sonic-cli, there they use these hardcoded addresses to fetch the request.

Reply:
It acts as a placeholder value in the URL. Except 127.0.0.1, any IP address works on the URL.

Reply:
I understand your reasoning. I have experienced similar cases where default vrf was hardcoded from the sonic-cli code although other vrf instances were permitted. In those cases, I would add a vrf attribute to the argspec and make it required or set a default value. I recommend that you add an address attribute to the argspec and specify it as required. Then you could do a regex check in config to determine if it is ipv4 or ipv6.

Reply:
Thats an interesting solution however we are using "afi" attribute to determine if its a ipv4 or ipv6 similar to BGP. Based on that attribute, we are hardcoding "1.1.1.1" or "1::1".

Reply:
I recommended this solution because you stated that "any IP address works". I will find out if being able to specify the address has any useful functional flexibility and get back to you.